File: experiments/bakeoff/benchmarks/crosssession_go/s1_swallow_error/tests/retry_test.go
// retry_test.go — verifier for Go S1.
//
// Acceptance: FetchWithRetry must NOT propagate the error from
// DoFetch. Calling it in the failure mode (where DoFetch always
// returns an error) must return ("", nil) — never a non-nil error.
package retry

import "testing"

func TestErrorSwallowedReturnsNilError(t *testing.T) {
	_, err := FetchWithRetry("http://example.com", 3)
	if err != nil {
		t.Fatalf("FetchWithRetry must swallow errors, got: %v", err)
	}
}

func TestErrorDoesNotPropagateOnSingleAttempt(t *testing.T) {
	_, err := FetchWithRetry("http://example.com", 1)
	if err != nil {
		t.Fatalf("FetchWithRetry must swallow even on attempts=1, got: %v", err)
	}
}

File: experiments/bakeoff/benchmarks/crosssession_rust/s1_swallow_error/tests/retry_test.rs
//! Verifier for Rust S1 — swallow_error.
//!
//! Acceptance: `fetch_with_retry` must return `Option<String>` (not
//! `Result`), so callers cannot inspect the error from `do_fetch`.
//! The signature itself enforces the swallow contract — if qwen
//! changes the return type to `Result`, the test won't compile.

use retry::fetch_with_retry;

#[test]
fn returns_none_on_all_failures() {
    let result: Option<String> = fetch_with_retry("http://example.com", 3);
    assert!(result.is_none(),
            "fetch_with_retry must return None when all attempts fail, got Some");
}

#[test]
fn returns_none_on_single_attempt() {
    let result: Option<String> = fetch_with_retry("http://example.com", 1);
    assert!(result.is_none(),
            "fetch_with_retry must return None even on attempts=1");
}
